exporter/splunkhecexporter/test/event_generator.go

```
@@ -0,0 +1,219 @@
+	fields := map[string]interface{}{
+		"metric_name:" + viperConfigVariable("config.METRIC_NAME"): 123,
+		"k0":                  "v0",
+		"k1":                  "v1",
+		"metric_name:cpu.usr": 11.12,
+		"metric_name:cpu.sys": 12.23,
+	}
+	jsonStr, err := json.Marshal(map[string]interface{}{
+		"index":      index,
+		"host":       "localhost",
+		"source":     source,
+		"sourcetype": sourcetype,
+		"event":      "metric",
+		"fields":     fields,
+	})
+	if err != nil {
+		panic(err)
+	}
+
+	requestBody := bytes.NewBuffer(jsonStr)
+	host := viperConfigVariable("config.METRICS_RECEIVER_URL")
+	req, err := http.NewRequest("POST", host, requestBody)
+	if err != nil {
+		panic(err)
+	}
+	req.Header.Set("Content-Type", "application/json")
+	client := &http.Client{}
+	resp, err := client.Do(req)
+	if err != nil {
+		panic(err)
+	}
+	defer resp.Body.Close()
+	bodyBytes, err := io.ReadAll(resp.Body)
+	if err != nil {
+		panic(err)
+	}
+	log.Printf("Send Metric Status Code: " + strconv.Itoa(resp.StatusCode) + "| Response Body: " + string(bodyBytes))
+	waitForEventToBeIndexedInSplunk()
+}
+
+func waitForEventToBeIndexedInSplunk(time_optional ...int) {
+	time_sec := 3
+	if len(time_optional) > 0 {
+		time_sec = time_optional[0]
+	}
+	time.Sleep(time.Duration(time_sec) * time.Second)
+}
+
+func addTextEvent(data string) {
+	path := viperConfigVariable("config.PATH_TO_OTEL_COLLECTOR_BIN_DIR")
+	f, err := os.OpenFile(path+"/test_file.json",
+		os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0666)
+	if err != nil {
+		log.Println(err)
+	}
+	defer f.Close()
+	if _, err := f.WriteString(data + "\n"); err != nil {
+		log.Println(err)
+	}
+	waitForEventToBeIndexedInSplunk(15)
+}
+
+func addJsonTextEvent(data map[string]interface{}) {
+	byteArray, _ := json.Marshal(data)
+	path := viperConfigVariable("config.PATH_TO_OTEL_COLLECTOR_BIN_DIR")
+	f, err := os.OpenFile(path+"/test_file.json",
+		os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0666)
+	if err != nil {
+		log.Println(err)
```

better to return the error and fail the test.